Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -125,6 +125,12 @@ extension PackageDependencyDescription {
             if isBaseURLRemote {
                 return url
             }
+            //
+            // If the dependency URL starts with '~/', try to expand it.
+            if url.hasPrefix("~/") {
+                return expandTilde(of: url)
```

Comment:
This can just be `fs.homeDirectory.appending(RelativePath(url.dropFirst().str))`

Reply:
We'll also need to pass the filesystem being used.